### PR TITLE
Ensures url has OK protocol.

### LINKED ISSUE #337: Clients hangs when trying to connect to a URL that is not NATS
Malformed URL throws exception but if you accidentally enter an URL that is valid but not nats-server the call hangs.

**Steps to reproduce:**

```
var cf = new ConnectionFactory();
var opts = ConnectionFactory.GetDefaultOptions()
opt.Servers = new[] {a valid URL to a web server e.g. http://localhost}
try
{
   var conn = cf.CreateConnection(opts)
}
catch(Exception ex)
{
   // Some Action
}
```
**Actual Result:**
Code hangs no exception thrown.
**Expected Result:**
Exception thrown

### PR DESCRIPTION
**Pull request by danielwertheim**
Closes #337 by checking for OK protocol.

### CODE REVIEW

File: src/NATS.Client/Options.cs [hunk 1]
```diff
@@ -247,6 +248,24 @@ internal Options(Options o)
...
+            if (string.IsNullOrWhiteSpace(url))
+                return url;
+            
+            if (url.StartsWith("nats://", StringComparison.OrdinalIgnoreCase))
+                return url;
+
+            if (url.StartsWith("tls://", StringComparison.OrdinalIgnoreCase))
+                return url;
+
+            var parts = url.Split(new[]{"://"}, StringSplitOptions.RemoveEmptyEntries);
```
Comment (watfordgnf) [inline src/NATS.Client/Options.cs]: Should hoist the allocation into a static.

**Review by watfordgnf — APPROVED** [review]
Small comment, otherwise looks good.

File: src/NATS.Client/Options.cs (hunk 1 above)
Comment (danielwertheim) [reply src/NATS.Client/Options.cs]: Fixed

**Review by ColinSullivan1 — APPROVED** [review]
LGTM - Thanks!

**Pull request merged**